webapi/src/test/scala/org/knora/webapi/store/triplestore/api/TriplestoreServiceInMemorySpec.scala

```diff
@@ -0,0 +1,318 @@
+          for {
+            result <- TriplestoreService.sparqlHttpSelect(query)
+          } yield assertTrue(
+            result == SparqlSelectResult(
+              SparqlSelectResultHeader(List("p", "o")),
+              SparqlSelectResultBody(
+                List(
+                  VariableResultsRow(
+                    Map(
+                      "p" -> "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
+                      "o" -> s"${Biblio.Class.JournalArticle.value}"
+                    )
+                  )
+                )
+              )
+            )
+          )
+        }
+      ),
+      suite("insertDataIntoTriplestore")(
+        test("given an empty list insertDataIntoTriplestore will insert the defauls") {
+          for {
+            _       <- TriplestoreService.insertDataIntoTriplestore(List.empty, prependDefaults = true)
+            ds      <- ZIO.serviceWithZIO[Ref[Dataset]](_.get)
+            contains = DefaultRdfData.data.map(_.name).map(namedModelExists(ds, _)).forall(_ == true)
+          } yield assertTrue(contains)
+        },
+        test("given an empty list insertDataIntoTriplestore will insert the defauls") {
+          for {
+            _ <- TriplestoreService.insertDataIntoTriplestore(
+                   List(
+                     RdfDataObject(
+                       path = "knora-ontologies/knora-base.ttl",
+                       name = "http://www.knora.org/ontology/knora-admin"
+                     )
+                   ),
+                   prependDefaults = false
+                 )
+            ds                <- ZIO.serviceWithZIO[Ref[Dataset]](_.get)
+            containsAdmin      = namedModelExists(ds, "http://www.knora.org/ontology/knora-admin")
+            doesNotContainBase = !namedModelExists(ds, "http://www.knora.org/ontology/knora-base")
+          } yield assertTrue(containsAdmin && doesNotContainBase)
+        }
+      ),
+      suite("sparqlHttpGraphFile")(test("sparqlHttpGraphFile should create the file") {
+        val tempDir = Files.createTempDirectory(UUID.randomUUID().toString)
+        tempDir.toFile.deleteOnExit()
+        val testFile = tempDir.toAbsolutePath.resolve("test.ttl")
+        ZIO.scoped {
+          for {
+            _ <- TriplestoreService.insertDataIntoTriplestore(
+                   List(
+                     RdfDataObject(
+                       path = "knora-ontologies/knora-base.ttl",
+                       name = "http://www.knora.org/ontology/knora-base"
+                     )
+                   ),
+                   prependDefaults = false
+                 )
+            _ <- TriplestoreService.sparqlHttpGraphFile("http://www.knora.org/ontology/knora-base", testFile, TriG)
+          } yield assertTrue({ val fileExists = Files.exists(testFile); fileExists })
+        }
+      })
+    ).provide(TriplestoreServiceInMemory.layer, datasetLayerFromTurtle(testDataSet), StringFormatter.test)
+
```

Comment:
I agree it would be nice and helpful, however currently the failing path is harder to understand and to get right to be the same as in the live implementation. By time I will extend these tests when integrating the impl in the ITs.